### DISCUSSION TITLE
New repo of an existing clone including history commits

### DISCUSSION THREAD

**Discussion by ioangkin** [community/community#22567] [New to GitHub]
<p>Hi, I am a completely new user, this is one of my first attempts to create a git repo and this is actually my first post in the forum; please excuse me if I use any wrong terminology or this is the wrong place to ask or my question has been answered elsewhere, unfortunately I couldn’t find a relevant answer.</p>
<p>The problem is that I have a local copy of a git repo that I am trying to clone in my current github account including any history commits performed in the original repo.</p>
<p>I am a final year student that worked for a company during my industrial placement year (as part of my studies) where I developed a web-app.</p>
<p>During the development I used the company’s internal version control system which is actually the github platform but used only internally in the company.</p>
<p>When I completed my placement I kept a copy of the whole code I developed including the git related files and folders (such as .git and .gitignore)</p>
<p>During my current graduating year I wish to use my github account to demonstrate the work I performed there and for this I am trying to clone the code folders/files as a repository in my github account. Also, I understand that the commits that I did during the development are recorded in the .git folder and I wonder whether there is any way these commits can be visible in the cloned repo.</p>
<p>I hope this question makes sense but please feel free to ask me for any additional information if needed.</p>
<p>Many thanks for any response.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>What you need to do is:</p>
<ol>
<li>Create a repository on GitHub by going to <a href="https://github.com/new" rel="nofollow noopener">https://github.com/new</a>
</li>
<li>Enter a name for your new repository</li>
<li>Select “Public” or “Private”</li>
<li>Click “Create repository”</li>
<li>Follow the instructions under the heading “…or push an existing repository from the command line”</li>
</ol>
<p>I hope that helps!</p>
<p>See the GitHub help docs <a href="https://help.github.com/articles/about-remote-repositories/" rel="nofollow noopener">about remote repositories</a>.</p>

**Comment by ioangkin** [comment]
<p>Thank you Lee, this helped.</p>
<p>The problem was that the local repository’s git settings and were pointing to the old (original) git address which is internal to the business and I don’t have access anymore.</p>
<p>To solve this I used GitHub Desktop to clone the local repository and changed the origin address under settings.</p>
<p>These are the steps I took for future reference:</p>
<ol>
<li>Create a repository on GitHub (as Lee suggested) and didn’t add any files (neither a readme)</li>
<li>Copy the new repository’s web address</li>
<li>Download and install GitHub Desktop</li>
<li>GitHub Desktop: File &gt; Clone local repository&gt; point to the local repository folder</li>
<li>GitHub Desktop: Paste the repository web address under: Repository &gt; repository Settings… &gt; Remote (tab) &gt; “Prime remote repository (origin) field”</li>
<li>GitHub Desktop: Repository &gt; Push (or Ctrl + P)</li>
</ol>
<p>This created an exact clone of my local repo to the new location in GitHub including all commit history with notes and proper dates. Although these commits are not marked as “confirmed” and do not count for my profile’s commits history.</p>
